code/ATMOSPHERICS/datum_pipe_network.dm

```diff
@@ -146,88 +129,44 @@
 		for(var/datum/gas_mixture/gas in gases)
 			var/volume_ratio = gas.volume / air_transient.volume
 
-			gas.oxygen = air_transient.oxygen * volume_ratio
-			gas.nitrogen = air_transient.nitrogen * volume_ratio
-			gas.toxins = air_transient.toxins * volume_ratio
-			gas.carbon_dioxide = air_transient.carbon_dioxide * volume_ratio
-
-			gas.temperature = air_transient.temperature
-
-			if(air_transient.trace_gases.len)
-				for(var/datum/gas/trace_gas in air_transient.trace_gases)
-					var/datum/gas/corresponding = locate(trace_gas.type) in gas.trace_gases
-
-					if(!corresponding)
-						corresponding = new trace_gas.type()
-						gas.trace_gases += corresponding
-
-					corresponding.moles = trace_gas.moles * volume_ratio
+			gas.copy_from(air_transient)
+			gas.multiply(volume_ratio)
 
 			gas.update_values()
 
 	air_transient.update_values()
 	return 1
 
-proc/equalize_gases(datum/gas_mixture/list/gases)
+proc/equalize_gases(list/datum/gas_mixture/gases)

```

Comment:
neither of these does anything useful unless something's changed, to byond it's all just ```list/gases```